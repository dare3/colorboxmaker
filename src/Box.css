/* Box.css */
.box {
    position: relative;
    margin: 10px;
    display: inline-block;
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  }
  
  .box div {
    width: 100px;
    height: 100px;
  }
  
  .remove-btn {
    position: absolute;
    top: -10px;
    right: -10px;
    background-color: red;
    color: white;
    border: none;
    border-radius: 50%;
    width: 25px;
    height: 25px;
    cursor: pointer;
  }
  