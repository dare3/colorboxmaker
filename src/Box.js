// Box.js
import './Box.css';
import React from 'react';

function Box({ id, width, height, backgroundColor, removeBox }) {
  const handleRemove = () => removeBox(id);

  return (
    <div className="box">
      <div 
        style={{ 
          width: `${width}px`, 
          height: `${height}px`, 
          backgroundColor: backgroundColor 
        }}
      />
      <button className="remove-btn" onClick={handleRemove}>X</button>
    </div>
  );
}

export default Box;
