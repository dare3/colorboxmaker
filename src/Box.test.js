import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

test('renders Box component', () => {
  const { asFragment } = render(<Box id="1" width="100" height="100" backgroundColor="blue" removeBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
