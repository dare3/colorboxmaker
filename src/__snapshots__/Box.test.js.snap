// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Box component 1`] = `
<DocumentFragment>
  <div
    class="box"
  >
    <div
      style="width: 100px; height: 100px; background-color: blue;"
    />
    <button
      class="remove-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
